### PR TITLE
Replace more GenericVector by std::vector and remove GenericVector::choose_nth_item

### PR DESCRIPTION
**Pull request by stweil**

### CODE REVIEW

File: src/ccstruct/blobs.cpp [hunk 1]
```diff
@@ -548,10 +548,10 @@ void TBLOB::GetEdgeCoords(const TBOX &box, std::vector<std::vector<int>> &x_coor
   y_coords.resize(box.width());
   CollectEdges(box, nullptr, nullptr, &x_coords, &y_coords);
   // Sort the output vectors.
-  for (int i = 0; i < x_coords.size(); ++i)
-    std::sort(x_coords[i].begin(), x_coords[i].end());
-  for (int i = 0; i < y_coords.size(); ++i)
-    std::sort(y_coords[i].begin(), y_coords[i].end());
+  for (auto coord : x_coords)
```
Comment (egorpugin) [inline src/ccstruct/blobs.cpp:551]: Copies are sorted, not original data. This is not the same as code before.

File: src/ccutil/unicharset.h [hunk 2]
```diff
@@ -307,9 +307,9 @@ class TESS_API UNICHARSET {
 
   // Delete CHAR_FRAGMENTs stored in properties of unichars array.
   void delete_pointers_in_unichars() {
-    for (int i = 0; i < unichars.size(); ++i) {
-      delete unichars[i].properties.fragment;
-      unichars[i].properties.fragment = nullptr;
+    for (auto unichar : unichars) {
```
Comment (egorpugin) [inline src/ccutil/unicharset.h:310]: Also not a reference.

File: src/ccstruct/normalis.cpp [hunk 3]
```diff
@@ -163,8 +162,8 @@ static void ComputeRunlengthImage(const TBOX &box,
   // Set a 2-d image array to the run lengths at each pixel.
   for (int ix = 0; ix < width; ++ix) {
     int y = 0;
-    for (int i = 0; i < y_coords[ix].size(); ++i) {
-      int y_edge = ClipToRange(y_coords[ix][i], 0, height);
+    for (auto y_coord : y_coords[ix]) {
```
Comment (egorpugin) [inline src/ccstruct/normalis.cpp:165]: If ClipToRange does not change first arg, copies may be expensive here (use `auto &`).

File: src/ccstruct/normalis.cpp [hunk 4]
```diff
@@ -182,8 +181,8 @@ static void ComputeRunlengthImage(const TBOX &box,
   // Now set the image pixels the the MIN of the x and y runlengths.
   for (int iy = 0; iy < height; ++iy) {
     int x = 0;
-    for (int i = 0; i < x_coords[iy].size(); ++i) {
-      int x_edge = ClipToRange(x_coords[iy][i], 0, width);
+    for (auto x_coord : x_coords[iy]) {
```
Comment (egorpugin) [inline src/ccstruct/normalis.cpp:184]: same as prev

File: src/ccstruct/blobs.cpp (hunk 1 above)
Comment (stweil) [reply src/ccstruct/blobs.cpp:551]: Right. I'll commit a fix.

File: src/ccstruct/normalis.cpp (hunk 3 above)
Comment (stweil) [reply src/ccstruct/normalis.cpp:165]: `y_coord` is of type `int`, to this should be fine.

File: src/ccstruct/normalis.cpp (hunk 4 above)
Comment (stweil) [reply src/ccstruct/normalis.cpp:184]: This is also an `int` value, so no need for a reference.

File: src/ccutil/unicharset.h (hunk 2 above)
Comment (stweil) [reply src/ccutil/unicharset.h:310]: Right. I'll commit a fix.

**Pull request merged**